src/wire/ipv6hop.rs

```diff
@@ -0,0 +1,307 @@
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+                               +
+// |                                                               |
+// .                                                               .
+// .                            Options                            .
+// .                                                               .
+// |                                                               |
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//
+//
+// See https://tools.ietf.org/html/rfc8200#section-4.3 for details.
+mod field {
+    #![allow(non_snake_case)]
+
+    use wire::field::*;
+
+    // Minimum size of the header.
+    pub const MIN_HEADER_SIZE: usize = 8;
+
+    // 8-bit identifier of the header immediately following this header.
+    pub const NXT_HDR:  usize = 0;
+    // 8-bit unsigned integer. Length of the OPTIONS field in 8-octet units,
+    // not including the first 8 octets.
+    pub const LENGTH:   usize = 1;
+    // Variable-length field. Option-Type-specific data.
+    //
+    // Length of the header is in 8-octet units, not including the first 8 octets. The first two
+    // bytes of the first octet are taken up by the next header type and the header length.
+    pub fn OPTIONS(length_field: u8) -> Field {
+        let bytes = length_field * 8 + 8;
+        2..bytes as usize
+    }
+}
+
+impl<T: AsRef<[u8]>> Packet<T> {
+    /// Create a raw octet buffer with an IPv6 Hop-by-Hop Options Header packet structure.
+    pub fn new(buffer: T) -> Packet<T> {
+        Packet { buffer }
+    }
+
+    /// Shorthand for a combination of [new] and [check_len].
+    ///
+    /// [new]: #method.new
+    /// [check_len]: #method.check_len
+    pub fn new_checked(buffer: T) -> Result<Packet<T>> {
+        let packet = Self::new(buffer);
+        packet.check_len()?;
+        Ok(packet)
+    }
+
+    /// Ensure that no accessor method will panic if called.
+    /// Returns `Err(Error::Truncated)` if the buffer is too short.
+    ///
+    /// The result of this check is invalidated by calling [set_header_len].
+    ///
+    /// [set_header_len]: #method.set_header_len
+    pub fn check_len(&self) -> Result<()> {
+        let data = self.buffer.as_ref();
+        let len = data.len();
+
+        if len < field::MIN_HEADER_SIZE {
+            return Err(Error::Truncated);
+        }
+
+        let of = field::OPTIONS(data[field::LENGTH]);
+
+        if len < of.end {
+            return Err(Error::Truncated);
+        }
+
+        Ok(())
+    }
+
+    /// Consume the packet, returning the underlying buffer.
+    pub fn into_inner(self) -> T {
+        self.buffer
+    }
+
+    /// Return the next header field.
+    #[inline]
+    pub fn next_header(&self) -> Protocol {
+        let data = self.buffer.as_ref();
+        Protocol::from(data[field::NXT_HDR])
+    }
+
+    /// Return the header length field. Length of the Hop-by-Hop Options header in 8-octet units,
```

Comment:
`/// Return the length of the Hop-by-Hop Options ...`

Reply:
fixed